src/portable/portable.h

```diff
@@ -31,7 +31,7 @@ typedef enum PortableChangeType {
         PORTABLE_WRITE,
         PORTABLE_MKDIR,
         _PORTABLE_CHANGE_TYPE_MAX,
-        _PORTABLE_CHANGE_TYPE_INVALID = INT_MIN,
+        _PORTABLE_CHANGE_TYPE_INVALID = -EINVAL,
```

Comment:
uh, hm, so this is problematic conceptually (though not in practice). PotableChangeType can be negative or positive, and we pick a _PORTABLE_CHANGE_TYPE_INVALID value really far away from the valid cases to ensure they don't conflict...

now in practice this doesn't matter, since EINVAL is 22, i.e. they do *not* conflict, but we should probably make this explicit, by mapping all negative values here explicitly, so that the compiler notices any collision

Reply:
sorry, actually it's more complicated... where we store PortableChangeType we can also store any errno actually, and suddenly it's even more messy

Reply:
I did not fully understand your fist message.
But, that is why, I did create separate commit for each change. I was not sure about all the changes, and the MR was merged very quickly.
Well we can just revert it, and add a comment about it.

Reply:
na, no need to revert. the change is good, but our enum use before, was a bit let's say… adventurous. I am working on a fix.